src/Okta.Idx.Sdk.UnitTests/IdxClientShould.cs

```diff
@@ -8746,5 +8746,1770 @@ public async Task SelectPhoneMethodTypeDuringChallenge()
+                        ""visible"": false,
+                        ""mutable"": false
+                    }
+                ],
+                ""accepts"": ""application/json; okta-version=1.0.0""
+            },
+            {
+                ""rel"": [
+                    ""create-form""
+                ],
+                ""name"": ""select-identify"",
+                ""href"": ""https://fake.example.com/idp/idx/identify/select"",
+                ""method"": ""POST"",
+                ""produces"": ""application/ion+json; okta-version=1.0.0"",
+                ""value"": [
+                    {
+                        ""name"": ""stateHandle"",
+                        ""required"": true,
+                        ""value"": ""02n8peIw_A9iBksPvgMSFrY7_nzepDvd8OHG_1jecT"",
+                        ""visible"": false,
+                        ""mutable"": false
+                    }
+                ],
+                ""accepts"": ""application/json; okta-version=1.0.0""
+            }
+        ]
+    },
+    ""cancel"": {
+        ""rel"": [
+            ""create-form""
+        ],
+        ""name"": ""cancel"",
+        ""href"": ""https://fake.example.com/idp/idx/cancel"",
+        ""method"": ""POST"",
+        ""produces"": ""application/ion+json; okta-version=1.0.0"",
+        ""value"": [
+            {
+                ""name"": ""stateHandle"",
+                ""required"": true,
+                ""value"": ""02n8peIw_A9iBksPvgMSFrY7_nzepDvd8OHG_1jecT"",
+                ""visible"": false,
+                ""mutable"": false
+            }
+        ],
+        ""accepts"": ""application/json; okta-version=1.0.0""
+    },
+    ""app"": {
+        ""type"": ""object"",
+        ""value"": {
+            ""name"": ""oidc_client"",
+            ""label"": ""Dotnet IDX Web App"",
+            ""id"": ""xxxxxxxx""
+        }
+    }
+}";
+            #endregion
+
+            MockHttpMessageHandler mockHttpMessageHandler = new MockHttpMessageHandler();
+            mockHttpMessageHandler.AddTestResponse("/oauth2/v1/interact", interactResponse);
+            mockHttpMessageHandler.AddTestResponse("/idp/idx/introspect", introspectResponse);
+            mockHttpMessageHandler.AddTestResponse("/idp/idx/enroll", enrollResponse);
+            mockHttpMessageHandler.AddTestResponse("/idp/idx/enroll/new", enrollNewResponse, System.Net.HttpStatusCode.BadRequest);
+
+            HttpClient httpClient = new HttpClient(mockHttpMessageHandler);
+
+            IdxClient idxClient = new IdxClient(TesteableIdxClient.DefaultFakeConfiguration, httpClient, NullLogger.Instance);
+            UserProfile userProfile = new UserProfile();
+            userProfile.SetProperty("firstName", "test-firstname");
+            userProfile.SetProperty("lastName", "test-lastname");
+            userProfile.SetProperty("email", "notagoodemail");
+
+            await Assert.ThrowsAsync<OktaApiException>(() => idxClient.RegisterAsync(userProfile));
```

Comment:
Same here, exception type should be `OktaException`.

Reply:
Let's assert the exception message too. You can do something like this:
```
            await function.Should()
                .ThrowAsync<OktaException>()
                .WithMessage("*<ExceptionMessage>*");
```

Reply:
See previous response.